cpp/src/sender.cpp

```diff
@@ -70,6 +71,10 @@ tracker sender::send(const message &message, const binary &tag) {
     pn_delivery_t *dlv = pn_delivery(
         pn_object(),
         pn_dtag((reinterpret_cast<const char *>(&tag[0])), tag.size()));
+
+    proton::binary tag_cp = tag;
+    tracker track = make_wrapper<tracker>(dlv);
+    send_span(message, tag_cp, track);
```

Comment:
Here you might want to use a pointer to message to allow you to copy the message when tracing is enabled and not copy the message when it isn't.
Something like
`proton::Message* message_to_send = &message;`

Reply:
Done, thanks! 

Also, we need to type cast it into a non-const pointer .
`proton::message* message_to_send = (proton::message*)&message;`

Reply:
This `const proton::message* message_to_send = &message;` also worked, no need to const cast. Thanks!